### PR TITLE
Single Group to Multi-Group Project Structure Migration

### PR DESCRIPTION
**Pull request by matthewgoodman13**
*Affliated with Issue #47*

*Description of changes:*
Followed [the kubebuilder migration document](https://book.kubebuilder.io/migration/multi-group.html) to migrate our project to support Multi-Group from our current Single Group structure.
- Moved all files under /api to /apis/{group_name} where our group_name for all our current files is multicluster
- Similarly, moved all files under controllers to controllers/multicluster
- Modified our PROJECT file to mark our project as a multi-group project
- Updated all imports that were affected by the above changes

*Testing:*
- `make test` completes successfully
- `make integration-suite` completes successfully
- `make run` with adding/removing services has all expected behavior

*Why?*
Need these changes to support additional groups to be added to our project such as `about.k8s.io` that contains the specs for `ClusterProperty`s.

### CODE REVIEW

**Review by runakash — COMMENTED** [review]
As you're renaming `pkg/api` to `pkg/apis`, there are maybe some places in the code you will have to update.

File: integration/scenarios/export_service.go [hunk 1]
```diff
@@ -8,7 +8,7 @@ import (
 	"time"
 
 	"github.com/aws/aws-cloud-map-mcs-controller-for-k8s/pkg/cloudmap"
-	"github.com/aws/aws-cloud-map-mcs-controller-for-k8s/pkg/controllers"
+	controllers "github.com/aws/aws-cloud-map-mcs-controller-for-k8s/pkg/controllers/multicluster"
```
[the remark this answers is not in the document]
Comment (matthewgoodman13) [reply integration/scenarios/export_service.go:11]: As per the [migration document](https://book.kubebuilder.io/migration/multi-group.html), all the controllers should be in a subdirectory of its group name, that is `multicluster` and hence the updates in path.

File: PROJECT [hunk 2]
```diff
@@ -3,6 +3,7 @@ layout:
 - go.kubebuilder.io/v3
 projectName: aws-cloud-map-mcs-controller-for-k8s
 repo: github.com/aws/aws-cloud-map-mcs-controller-for-k8s
+multigroup: true
 resources:
 - api:
```
[the remark this answers is not in the document]
Comment (matthewgoodman13) [reply PROJECT:8]: Resolved.
Path updated to:
`path: github.com/aws/aws-cloud-map-mcs-controller-for-k8s/apis/multicluster/v1alpha1`

File: integration/scenarios/export_service.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (runakash) [reply integration/scenarios/export_service.go:11]: Do we plan to have separate controller for clusterId?

[the remark this answers is not in the document]
Comment (matthewgoodman13) [reply integration/scenarios/export_service.go:11]: To be discussed in the design meeting. I do argue that having a separate controller for clusterproperty attributes are beneficial just to separate the responsibilities of the controllers. A ClusterPropertyController can be responsible for creating the ClusterProperty for `id.k8s.io` and then we can still have watchers for any changes to that object

[the remark this answers is not in the document]
Comment (matthewgoodman13) [reply integration/scenarios/export_service.go:11]: In the above scenario, we would need another "group" for `about` (for the about.k8s.io ClusterProperty)

File: pkg/controllers/multicluster/cloudmap_controller_test.go [hunk 3]
```diff
@@ -6,7 +6,7 @@ import (
 	"testing"
 
 	cloudmapMock "github.com/aws/aws-cloud-map-mcs-controller-for-k8s/mocks/pkg/cloudmap"
-	"github.com/aws/aws-cloud-map-mcs-controller-for-k8s/pkg/api/v1alpha1"
+	"github.com/aws/aws-cloud-map-mcs-controller-for-k8s/pkg/apis/multicluster/v1alpha1"
```
[the remark this answers is not in the document]
Comment (matthewgoodman13) [reply pkg/controllers/multicluster/cloudmap_controller_test.go:9]: I agree, already been starting to do that due to conflicts.
Have been updating `controllers` to `multiclustercontrollers` and `aboutcontrollers` when running into conflicts. 
Similarly for the second, I've started to use `multiclusterv1alpha1` and `aboutv1alpha1`.

Can refactor this everywhere if we like my format. They are also getting a bit long, so also open for suggestions. 
Once we agree on a new format, will update this PR.

**Review by astaticvoid — APPROVED** [review]

**Review by runakash — APPROVED** [review]

**Pull request merged**